api/v1alpha1/clusterextension_types.go

```diff
@@ -46,6 +46,78 @@ const (
 
 // ClusterExtensionSpec defines the desired state of ClusterExtension
 type ClusterExtensionSpec struct {
+	// source is the selected resolution source such as catalog or bundle
```

Comment:
As per the OpenShift API Conventions we are following, the GoDoc should be pretty terse and allow a user to answer the following questions from reading it:

- What is the purpose of this field? What does it allow me to achieve?
- How does this field interact with other fields or features?
- What are the limitations of this field?
- Is this field optional or required?
    - IF optional, what happens if the field is omitted?

Reply:
@everettraven Please review these latest docs changes at your convenience (tomorrow, of course).